packages/core/src/shared/sam/build.ts

```diff
@@ -233,16 +237,28 @@ export async function runBuild(arg?: TreeNode): Promise<SamBuildResult> {
     await updateRecentResponse(buildMementoRootKey, 'global', 'templatePath', templatePath)
 
     try {
-        // Create a child process to run the SAM build command
-        const buildProcess = new ChildProcess(samCliPath, ['build', ...buildFlags], {
-            spawnOptions: await addTelemetryEnvVar({
-                cwd: params.projectRoot.fsPath,
-                env: await getSpawnEnv(process.env),
-            }),
-        })
+        await vscode.window.withProgress(
+            {
+                cancellable: true,
+                location: vscode.ProgressLocation.Notification,
+            },
+            async (progress) => {
+                progress.report({ message: `Building SAM template at ${params.template.uri.path}` })
```

Comment:
if you make this cancellable I think you should also be adding the additional token argument.

e.g.
```
async (progress, token) => {
    token.onCancellationRequested(async () => {
         // do something
         throw new CancellationError('user')
    })
}
```

Reply:
Updated